#!/usr/bin/env python
PACKAGE = "gait"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("stand_offset_x", double_t, 0, "Foothold offset to the projection of the shoulder", 0, -0.05, 0.05)
gen.add("stand_offset_y", double_t, 0, "Foothold offset to the projection of the shoulder", -0.02, -0.05, 0.05)
gen.add("duty_factor", double_t, 0, "duration of support phase over a leg cycle", 0.75, 0.75, 1)
gen.add("leg_cycle", double_t, 0, "duration of a leg cycle in sec", 5, 2, 10)
gen.add("step_height", double_t, 0, "height(metre) of foot trajectory", 0.03, 0.01, 0.05)
gen.add("cmd_vel_timeout", double_t, 0, "how long the vel_cmd msg will expire, in secs", 3, 0.25, 10)

exit(gen.generate(PACKAGE, "walk gait", "Walk"))